src/App.js

```diff
@@ -1,53 +1,77 @@
-import React, { useState, useEffect } from 'react';
-import Header from './components/Header';
-import Vacations from './components/Vacations';
-import './App.css';
-import axios from 'axios';
+import React, { useState, useEffect } from "react";
+import Header from "./components/Header";
+import Vacations from "./components/Vacations";
+import Notes from "./components/Notes";
+import Home from "./components/Home";
+import "./App.css";
+import axios from "axios";
+import { getHash } from "./utils";
+import qs from "qs";
 
-const API = 'https://acme-users-api-rev.herokuapp.com/api';
+const API = "https://acme-users-api-rev.herokuapp.com/api";
 
 //FETCHES RANDOM USER ON LOAD
 const fetchUser = async () => {
   const storage = window.localStorage;
-  const userId = storage.getItem('userId');
+  const userId = storage.getItem("userId");
   if (userId) {
     try {
       return (await axios.get(`${API}/users/detail/${userId}`)).data;
     } catch (ex) {
-      storage.removeItem('userId');
+      storage.removeItem("userId");
       return fetchUser();
     }
   }
   const user = (await axios.get(`${API}/users/random`)).data;
-  storage.setItem('userId', user.id);
+  storage.setItem("userId", user.id);
   return user;
 };
 
 //FETCHES VACATION LIST BASED ON USER
 const fetchVacations = async () => {
   const storage = window.localStorage;
-  const userId = storage.getItem('userId');
+  const userId = storage.getItem("userId");
   const vacations = (await axios.get(`${API}/users/${userId}/vacations`)).data;
   return vacations;
 };
 
+//FETCHES NOTES BASED ON USER
+const fetchNotes = async () => {
+  const storage = window.localStorage;
+  const userId = storage.getItem("userId");
+  const notes = (await axios.get(`${API}/users/${userId}/notes`)).data;
+  return notes;
+};
+
 function App() {
-  //USER AND VACATION LIST STATES
+  //USER, VACATION, AND NOTES STATES
   const [user, setUser] = useState({});
   const [vacations, setVacations] = useState([]);
+  const [notes, setNotes] = useState([]);
+  //ROUTING/PARAMS STATE
+  const [params, setParams] = useState(qs.parse(getHash()));
 
-  //ASSIGNS USER AND VACATION LIST TO STATE
+  //ASSIGNS USER TO STATE AND CREATES HASHCHANGE LISTNER
   useEffect(() => {
+    window.addEventListener("hashchange", () => {
+      setParams(qs.parse(getHash()));
+    });
+    setParams(qs.parse(getHash()));
     fetchUser().then(data => setUser(data));
+  }, []);
+
+  //ASSIGNS NOTES AND VACATION LIST TO STATE
```

Comment:
I removed the setUser from this useEffect because the dependency array included user. Everytime the user updated the useEffect would run and update the user over and over again.